geode-gfsh/src/test/java/org/apache/geode/management/internal/cli/commands/ConnectCommandTest.java

```diff
@@ -417,4 +417,21 @@ public void connectToManagerBefore1_10() {
         .statusIsError()
         .containsOutput("Cannot use a 1.14 gfsh client to connect to a 1.9 cluster");
   }
+
+  @Test
+  public void connectToManagerBySerializationVersion() {
+    when(gfsh.getVersion()).thenReturn("0.0.0");
+    when(gfsh.getGeodeSerializationVersion()).thenReturn("1.14.0");
+    when(operationInvoker.getRemoteVersion()).thenReturn("0.0.0");
+    when(operationInvoker.getRemoteGeodeSerializationVersion()).thenReturn("1.14.0");
+    when(operationInvoker.isConnected()).thenReturn(true);
+
+    ResultModel resultModel = new ResultModel();
+    when(connectCommand.jmxConnect(any(), anyBoolean(), any(), any(), anyBoolean()))
+        .thenReturn(resultModel);
+
+    gfshParserRule.executeAndAssertThat(connectCommand, "connect --locator=localhost:4040")
+        .statusIsSuccess()
+        .doesNotContainOutput("Cannot use a 0.0.0 gfsh client to connect to a 0.0.0 cluster");
+  }
```

Comment:
This is my understanding of the equivalence classes for testing are (rsv = remote serialization version read via JMX, rmv = remote marketing version read via JMX)

```
rsv >= 1.12 && rmv == dont-care
   compatible, covered by connectToManagerBySerializationVersion()
rsv < 1.12
   impossible
rsv unknown
   rmv >= 1.10
     compatible, covered by connectToManagerOlderThan1_10() sic
   rmv < 1.10
      incompatible
   rmv unknown
      incompatible
```

By the way `connectToManagerOlderThan1_10()` is mis-named: it's actually connecting to version 1.10—not a version _older_ than 1.10. Please fix this name.

But then I also see these tests:

```
connectToManagerWithDifferentMajorVersion()
connectToManagerWithDifferentMinorVersion()
```

So major and minor versions seem to be important too. But only for certain version ranges? Actually `ConnectCommand` does not, in general respect semver at all. I think those tests need to be removed.

Reply:
The different-minor-version test was a holdover from the old days when gfsh was locked to same major&minor as the cluster.  I removed it.  I think major versions still play a role, as we need to carefully consider how far into the future we are willing to guarantee forward compatibility.  It seems safest for now to presume that gfsh 1.x won't be interoperable with a future Geode 2.x.  The reverse might be more plausible but we can unlock backward major version compatibility if and when the time comes.